lib/travis/build/config.rb

```diff
@@ -33,14 +53,30 @@ def go_version_aliases_hash
           trusty: ENV.fetch('TRAVIS_BUILD_APT_SOURCE_WHITELIST_TRUSTY', '')
         },
         apt_whitelist_skip: ENV.fetch('TRAVIS_BUILD_APT_WHITELIST_SKIP', ''),
+        cabal_default: ENV.fetch('TRAVIS_BUILD_CABAL_DEFAULT', '1.22'),
         enable_debug_tools: ENV.fetch(
           'TRAVIS_BUILD_ENABLE_DEBUG_TOOLS',
           ENV.fetch('TRAVIS_ENABLE_DEBUG_TOOLS', '')
         ),
         etc_hosts_pinning: ENV.fetch(
           'TRAVIS_BUILD_ETC_HOSTS_PINNING', ENV.fetch('ETC_HOSTS_PINNING', '')
         ),
-        ghc_default: ENV.fetch('TRAVIS_BUILD_GHC_DEFAULT', '7.8.4'),
+        ghc_default: ENV.fetch('TRAVIS_BUILD_GHC_DEFAULT', '7.10.3'),
+        ghc_version_aliases: ENV.fetch(
+          'TRAVIS_BUILD_GHC_VERSION_ALIASES', (
+            latest_semver_aliases(
```

Comment:
technically ghc versions aren't semver, if anything they are probably pvp. It's really hard to write non-trivial code which would work with say `ghc == 8.x.x`, as `8.2` will definitely break something.

I.e. I'd remove single version alias all-together, or actively discourage from using such.